site/profile/manifests/android.pp

```diff
@@ -0,0 +1,33 @@
+
+class profile::android (
+        String $sdk_tools,
+        String $sdk_path,
+	Array[String] $npm_packages,
+	Array[String] $sdk_packages,
+)
+
+{
+  archive { "/tmp/$sdk_tools":
+    source       => "https://dl.google.com/android/repository/$sdk_tools",
+    extract      => true,
+    extract_path => $sdk_path,
+    user          => 'ubuntu',
+    group         => 'ubuntu',
+    }
+  $npm_packages.each |String $npm_package| {
+    exec { "install_global_npm_modules_${npm_package}":
+      command  => "sudo npm i -g \"${npm_package}\"",
+    }
+  }
+  exec { "Set ANDROID_HOME":
+    command => "sed -e '/ANDROID_HOME/d' -i /etc/environment && sudo sh -c \"echo \"ANDROID_HOME=\"${sdk_path}\"\" >> /etc/environment\"",
```

Comment:
use creates, unless or onlyif parameters to avoid exec command to be run every time.

Reply:
There are better ways to manage line contents in files. like the resource ensure_line

Reply:
Use something like:

file_line { "foo_env_var":
    ensure  => present,
    line    => "Foo=${bar}",
    path    => "/etc/environment",
}